impl_core/src/test/java/io/opencensus/implcore/tags/propagation/CorrelationContextFormatTest.java

```diff
@@ -0,0 +1,294 @@
+    assertThat(carrier).isEmpty();
+    tagsComponent.setState(TaggingState.ENABLED);
+  }
+
+  @Test
+  public void inject_TooManyTags() throws TagContextSerializationException {
+    Tag[] tags = new Tag[CorrelationContextFormat.MAX_NUMBER_OF_TAGS + 1];
+    for (int i = 0; i < tags.length; i++) {
+      tags[i] =
+          Tag.create(
+              TagKey.create("k" + i), TagValue.create("v" + i), METADATA_UNLIMITED_PROPAGATION);
+    }
+    TagContext tagContext = makeTagContext(tags);
+    Map<String, String> carrier = new HashMap<String, String>();
+    thrown.expect(TagContextSerializationException.class);
+    textFormat.inject(tagContext, carrier, setter);
+  }
+
+  @Test
+  public void inject_SizeTooLarge() throws TagContextSerializationException {
+    Tag[] tags = new Tag[40];
+    for (int i = 0; i < tags.length; i++) {
+      tags[i] =
+          Tag.create(
+              TagKey.create(generateRandom(240)),
+              TagValue.create(generateRandom(240)),
+              METADATA_UNLIMITED_PROPAGATION);
+    }
+    TagContext tagContext = makeTagContext(tags);
+    Map<String, String> carrier = new HashMap<String, String>();
+    thrown.expect(TagContextSerializationException.class);
+    textFormat.inject(tagContext, carrier, setter);
+  }
+
+  @Test
+  public void extract() throws TagContextDeserializationException {
+    Map<String, String> carrier = Collections.singletonMap(CORRELATION_CONTEXT, "k1=v1,k2=v2");
+    TagContext tagContext = textFormat.extract(carrier, getter);
+    assertThat(TagsTestUtil.tagContextToList(tagContext)).containsExactly(T1, T2);
+  }
+
+  @Test
+  public void extract_Empty() throws TagContextDeserializationException {
+    Map<String, String> carrier = Collections.singletonMap(CORRELATION_CONTEXT, "");
+    TagContext tagContext = textFormat.extract(carrier, getter);
+    assertThat(TagsTestUtil.tagContextToList(tagContext)).isEmpty();
+  }
+
+  @Test
+  public void extract_WithUnknownProperties() throws TagContextDeserializationException {
+    Map<String, String> carrier =
+        Collections.singletonMap(CORRELATION_CONTEXT, "k1=v1;property1=p1;property2=p2,k2=v2");
+    Tag expected = Tag.create(K1, TagValue.create("v1"), METADATA_UNLIMITED_PROPAGATION);
+    TagContext tagContext = textFormat.extract(carrier, getter);
+    assertThat(TagsTestUtil.tagContextToList(tagContext)).containsExactly(expected, T2);
+  }
+
+  @Test
+  public void extract_WithSpaces() throws TagContextDeserializationException {
+    Map<String, String> carrier = Collections.singletonMap(CORRELATION_CONTEXT, "k1= v1, k2=v 2");
+    Tag expected1 = Tag.create(K1, TagValue.create(" v1"), METADATA_UNLIMITED_PROPAGATION);
+    Tag expected2 =
+        Tag.create(TagKey.create(" k2"), TagValue.create("v 2"), METADATA_UNLIMITED_PROPAGATION);
+    TagContext tagContext = textFormat.extract(carrier, getter);
+    assertThat(TagsTestUtil.tagContextToList(tagContext)).containsExactly(expected1, expected2);
+  }
```

Comment:
In w3c [spec](https://github.com/w3c/correlation-context/blob/master/correlation_context/HTTP_HEADER_FORMAT.md#properties) it say
"Spaces are allowed between properties and before and after equal sign. Properties with spaces MUST be considered identical to properties with all spaces trimmed.
"
Above implies that TagKey(" k1", " v1") = TagKey("k1","v1"). If so, it will warrant another test where you already have TagKey("k1", "v1 old"), and adding TagKey(" k1", "v1 new") should update the old one rather than create a new Tag.

Reply:
You're right. Since in OpenCensus we allow spaces in tags, and spaces are treated as part of the key/value, I modified the decoding to trim the string keys/values.